pkgs/applications/editors/emacs/generic.nix

```diff
@@ -337,6 +340,9 @@ mkDerivation (finalAttrs: (lib.optionalAttrs withNativeCompilation {
   ++ lib.optional withXwidgets "--with-xwidgets"
   ;
 
+  CFLAGS = lib.optional (variant == "macport")
```

Comment:
Put this inside `env`.
Also, use `NIX_CFLAGS_COMPILE`

Reply:
P.S.: is this limited to MacPort, or it should affect mainline-for-macos too?

Reply:
It is better to use `NIX_CFLAGS_COMPILE` (inside `env = {. . .}`) because `NIX_` env-vars are is in a higher level.